charts/xp-management-service/templates/_helpers.tpl

```diff
@@ -0,0 +1,181 @@
+*/}}
+{{- define "management-svc.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "management-svc.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "management-svc.serviceAccountName" -}}
+{{- if .Values.deployment.serviceAccount.create }}
+{{- default (include "management-svc.fullname" .) .Values.deployment.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.deployment.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{- define "management-svc.environment" -}}
+{{- .Values.global.environment | default .Values.deployment.environment | default "dev" -}}
+{{- end -}}
+
+{{- define "management-svc.sentry.enabled" -}}
+{{ eq ((.Values.deployment.apiConfig.sentryConfig).enabled | toString) "true" }}
+{{- end -}}
+
+{{- define "management-svc.sentry.dsn" -}}
+{{- .Values.global.sentry.dsn | default .Values.deployment.sentry.dsn -}}
+{{- end -}}
+
+{{- define "management-svc.ui.defaultConfig" -}}
+{{- if .Values.uiConfig -}}
+appConfig:
+  environment: {{ .Values.uiConfig.appConfig.environment | default (include "management-svc.environment" .) }}
+authConfig:
+  oauthClientId: {{ .Values.global.oauthClientId | default .Values.uiConfig.authConfig.oauthClientId | quote }}
+{{- if (include "management-svc.sentry.enabled" .) }}
+sentryConfig:
+  environment: {{ .Values.uiConfig.sentryConfig.environment | default (include "management-svc.environment" .) }}
+  dsn: {{ .Values.uiConfig.sentryConfig.dsn | default (include "management-svc.sentry.dsn" .) | quote }}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "management-svc.ui.config" -}}
+{{- $defaultConfig := include "management-svc.ui.defaultConfig" . | fromYaml -}}
+{{ .Values.uiConfig | merge $defaultConfig | toPrettyJson }}
+{{- end -}}
+
+{{/*
+API config related
+*/}}
+
+{{- define "management-svc.defaultConfig" -}}
+port: 8080
+allowedOrigins: "*"
+authorizationConfig:
+  enabled: false
+dbConfig:
+  host: {{ include "xp-postgresql.host" . | quote }}
+  port: 5432
+  database:  {{ include "xp-postgresql.database" . }}
+  user:  {{ include "xp-postgresql.username" . }}
+  connMaxIdleTime: {{ .Values.deployment.apiConfig.dbConfig.connMaxIdleTime }}
+  connMaxLifetime: {{ .Values.deployment.apiConfig.dbConfig.connMaxLifetime }}
+  maxIdleConns: {{ .Values.deployment.apiConfig.dbConfig.maxIdleConns }}
+  maxOpenConns: {{ .Values.deployment.apiConfig.dbConfig.maxOpenConns }}
+deploymentConfig:
+  dnvironmentType: dev
+segmenterConfig:
+  s2_ids:
+    minS2CellLevel: 10
+    maxS2CellLevel: 14
+mlpConfig:
+  url: http://mlp:8080/v1
+newRelicConfig:
+  enabled: false
+sentryConfig:
+  enabled: false
+xpUIConfig:
+  appDirectory: /app/xp-ui
+{{- end -}}
+
+{{- define "management-svc.config" -}}
+{{- $defaultConfig := include "management-svc.defaultConfig" . | fromYaml -}}
+{{ .Values.deployment.apiConfig | merge $defaultConfig | toYaml }}
+{{- end -}}
+
+{{/*
+Postgres related
```

Comment:
No need to define these here anymore, You can use helper functions from the common chart. 

ref: https://github.com/caraml-dev/helm-charts/blob/main/charts/mlp/templates/configmap.yaml#L25